pyglet/graphics/shader.py

```diff
@@ -236,31 +236,128 @@ def __repr__(self):
         return f"Attribute(name='{self.name}', location={self.location}, count={self.count})"
 
 
+class _UniformArray:
+    """Wrapper of the GLSL array data inside a Uniform.
+    Allows access to get and set items for a more Pythonic implementation.
+    Types with a length longer than 1 will be returned as tuples as an inner list would not support individual value
+    reassignment. Array data must either be set in full, or by indexing."""
+
+    _slots_ = ('uniform', 'gl_type', 'gl_getter', 'gl_setter', 'is_matrix', 'dsa', '_c_array', '_ptr')
+    def __init__(self, uniform, gl_getter, gl_setter, gl_type, is_matrix, dsa):
+        self._uniform = uniform
+        self._gl_type = gl_type
+        self._gl_getter = gl_getter
+        self._gl_setter = gl_setter
+        self._is_matrix = is_matrix
+        self._dsa = dsa
+
+        if self._uniform.length > 1:
```

Comment:
Assuming `_uniform.length` is always 1 for "normal" uniforms, could we skip the `if` check and always just do: ?
```py
self._c_array = (gl_type * self._uniform.length * self._uniform.size)()
```
The result should be the same, and we can avoid the if/else cost.

Reply:
Unfortunately not, by doing `gl_type * self._uniform.length * self._uniform.size` it will create an array of an array even if length is 1. Which is fine for > 1, but if you want an array of floats for instance, then all of your floats are going to be sublists or tuples with only 1 value, which isn't very user friendly if we want to make it behave more like a list.

Reply:
Ah, you're right. I forgot about that ctypes multiplication behavior.